tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_attestation.py

```diff
@@ -79,3 +83,31 @@ def test_invalid_nonset_committe_bits(spec, state):
     attestation.committee_bits[committee_index] = 0
 
     yield from run_attestation_processing(spec, state, attestation, valid=False)
+
+
+@with_electra_and_later
+@spec_state_test
+@with_presets([MINIMAL], "need multiple committees per slot")
+@always_bls
+def test_multiple_committees(spec, state):
+    attestation_data = build_attestation_data(spec, state, slot=state.slot, index=0)
+    attestation = spec.Attestation(data=attestation_data)
+
+    # fill the attestation with two committees
+    fill_aggregate_attestation(spec, state, attestation, signed=True, committee_index=0)
+    fill_aggregate_attestation(spec, state, attestation, signed=True, committee_index=1)
+
+    next_slots(spec, state, spec.MIN_ATTESTATION_INCLUSION_DELAY)
+
+    yield from run_attestation_processing(spec, state, attestation)
+
+
+@with_electra_and_later
+@spec_state_test
+@with_presets([MINIMAL], "need multiple committees per slot")
+@always_bls
+def test_one_committee_with_gap(spec, state):
+    attestation = get_valid_attestation(spec, state, index=1, signed=True)
+    next_slots(spec, state, spec.MIN_ATTESTATION_INCLUSION_DELAY)
+
+    yield from run_attestation_processing(spec, state, attestation)
```

Comment:
1. Let's move these tests to phase0?
2. For `test_multiple_committees`, I believe we have similar tests in phase0 sanity or fork choice that cover this case. But fine to add it. 👍

Reply:
These tests make sense for Electra because an attestation can have more than one committee. Do you think they are applicable for pre-Electra forks? If yes, I am more than happy to move them.

> For `test_multiple_committees`, I believe we have similar tests in phase0 sanity or fork choice that cover this case. But fine to add it. 👍

During a quick look I haven’t found one, but realized that a subsequent call to `fill_aggregate_attestation` resets the aggregation bits to zero which isn’t a desirable test scenario. I will work on to fix this

Reply:
Yes, indeed, there is a `state_transition_with_full_block` that is called from some FC tests, and in Electra it will produce a block containing an on chain attestation of a new type with all committees in a single bitfield. I think it is still makes sense to have a separate state test covering this case, so decided to submit a fix rather than remove it